core/src/components/checkbox/checkbox.tsx

```diff
@@ -174,6 +189,48 @@ export class Checkbox implements ComponentInterface {
     this.toggleChecked(ev);
   };
 
+  private getHintTextID(): string | undefined {
+    const { el, helperText, errorText, helperTextId, errorTextId } = this;
+
+    if (el.classList.contains('ion-touched') && el.classList.contains('ion-invalid') && errorText) {
+      return errorTextId;
+    }
+
+    if (helperText) {
+      return helperTextId;
+    }
+
+    return undefined;
+  }
+
+  /**
+   * Responsible for rendering helper text and error text.
+   * This element should only be rendered if hint text is set.
+   */
+  private renderHintText() {
+    const { helperText, errorText, helperTextId, errorTextId } = this;
+
+    /**
+     * undefined and empty string values should
+     * be treated as not having helper/error text.
+     */
+    const hasHintText = !!helperText || !!errorText;
+    if (!hasHintText) {
```

Comment:
why create the const and not make the validation on the if directly?

Reply:
I just copied this implementation from input: https://github.com/ionic-team/ionic-framework/blob/cd5c27a12a03cc11da7ae86d89ad402343fed1c0/core/src/components/input/input.tsx#L625